File: python/tts.py
from google.cloud import texttospeech

# 클라이언트 생성
client = texttospeech.TextToSpeechClient()

# 변환할 텍스트 입력
synthesis_input = texttospeech.SynthesisInput(text="Hello, World!")

# 목소리 언어, 성별 설정
voice = texttospeech.VoiceSelectionParams(
    language_code="en-US", ssml_gender=texttospeech.SsmlVoiceGender.NEUTRAL
)

# 오디오 파일 설정
audio_config = texttospeech.AudioConfig(
    audio_encoding=texttospeech.AudioEncoding.LINEAR16
)

# 텍스트, 목소리, 오디오를 포함하여 요청을 보내고 응답을 받음
response = client.synthesize_speech(
    input=synthesis_input, voice=voice, audio_config=audio_config
)

# 오디오 파일 생성
with open("output.mp3", "wb") as out:
    # Write the response to the output file.
    out.write(response.audio_content)
    print('Audio content written to file "output.mp3"')

File: TTS/java/src/main/java/org/example/TTS.java
package org.example;

import com.google.cloud.texttospeech.v1.*;
import com.google.protobuf.ByteString;

import java.io.FileOutputStream;
import java.io.OutputStream;

public class TTS {

    public static void main(String... args) throws Exception {
        System.out.println(System.getenv("GOOGLE_APPLICATION_CREDENTIALS"));

        // 클라이언트 연결하기
        try (TextToSpeechClient textToSpeechClient = TextToSpeechClient.create()) {
            // 변환할 텍스트
            SynthesisInput input = SynthesisInput.newBuilder().setText("Hello, World!").build();

            // 목소리의 언어와 성별 등을 설정
            VoiceSelectionParams voice =
                    VoiceSelectionParams.newBuilder()
                            .setLanguageCode("en-US")
                            .setSsmlGender(SsmlVoiceGender.NEUTRAL)
                            .build();

            // 오디오 파일 설정
            AudioConfig audioConfig =
                    AudioConfig.newBuilder().setAudioEncoding(AudioEncoding.LINEAR16).build();

            // text, 설정으로 요청 보내기
            SynthesizeSpeechResponse response =
                    textToSpeechClient.synthesizeSpeech(input, voice, audioConfig);

            // 응답
            ByteString audioContents = response.getAudioContent();

            // 해당 응답을 오디오 파일로 만들기
            try (OutputStream out = new FileOutputStream("output.mp3")) {
                out.write(audioContents.toByteArray());
                System.out.println("Audio content written to file \"output.mp3\"");
            }
        }
    }
}